components/login/SignInKey.tsx

```diff
@@ -115,4 +132,27 @@ const SignInKey = (props: LoginStageProps): React.ReactElement => {
   );
 };
 
+function buildLockoutErrorMessage(
+  t: TFunction,
+  data: Record<string, number>
+): { title: string; description: string; displaySpecificErrors: boolean } {
+  if (data.remainingNumberOfAttemptsBeforeLockout) {
+    return {
+      title: t("loginInvalidCredentialsErrorTitle"),
+      description: `${t("remainingNumberOfLoginAttemptsErrorMessagPart1")} ${
+        data.remainingNumberOfAttemptsBeforeLockout
+      } ${t("remainingNumberOfLoginAttemptsErrorMessagPart2")}`,
+      displaySpecificErrors: true,
+    };
+  } else {
+    return {
+      title: t("lockoutErrorTitle"),
+      description: `${t("lockoutDurationErrorMessagePart1")} ${
+        data.numberOfSecondsBeforeLockoutExpires
```

Comment:
Does there need to be any check to ensure that `numberOfSecondsBeforeLockoutExpires` exists?

Reply:
No, because if `numberOfSecondsBeforeLockoutExpires` is undefined a TypeError/Application Error will not be thrown.